power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -816,6 +904,28 @@ class Deserializer {
         }
     }
 
+    static void set_nan(row_based_t /*tag*/, Buffer const& buffer, ComponentInfo const& info) {
+        assert(buffer.data != nullptr);
+        info.component->set_nan(buffer.data, 0, info.total_elements);
+    }
+    static void set_nan(columnar_t /*tag*/, Buffer const& buffer, ComponentInfo const& info) {
+        assert(buffer.data == nullptr);
+        for (auto const& attribute_buffer : buffer.attributes) {
+            if (attribute_buffer.meta_attribute != nullptr) {
+                ctype_func_selector(attribute_buffer.meta_attribute->ctype, [&attribute_buffer, &info]<typename T> {
+                    std::ranges::fill(std::span{reinterpret_cast<T*>(attribute_buffer.data),
+                                                narrow_cast<size_t>(info.total_elements)},
+                                      nan_value<T>);
+                });
+            }
+        }
+    }
+
+    static BufferView advance(BufferView buffer, Idx offset) {
```

Comment:
Now I see your point here about copying of the `std::vector<AttributeBuffer<void>>`. In the `advance` you made implicit copy of `BufferView` which copies the whole `reordered_attribute_buffers`. This `advance` is called for every scenario and every element, which indeed make it very expensive copy. Maybe we still need to change that....